network/conn.go

```diff
@@ -0,0 +1,58 @@
+package network
+
+import (
+	"io"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	ic "github.com/libp2p/go-libp2p-crypto"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// Conn is a connection to a remote peer. It multiplexes streams.
+// Usually there is no need to use a Conn directly, but it may
+// be useful to get information about the peer on the other side:
+//  stream.Conn().RemotePeer()
+type Conn interface {
+	io.Closer
+
+	ConnSecurity
+	ConnMultiaddrs
+
+	// NewStream constructs a new Stream over this conn.
+	NewStream() (Stream, error)
+
+	// GetStreams returns all open streams over this conn.
+	GetStreams() []Stream
```

Comment:
does Conn work with streams? I thought only MuxedConn does

Reply:
These methods do indeed seem a bit redundant. Adding it to iteration 3 for potential refactor.